// Copyright 2019 Google Cloud Platform Proxy Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.api.envoy.http.backend_auth;

import "validate/validate.proto";

message BackendAuthRule {
    // Operation name.
    string operation = 1 [(validate.rules).string.min_bytes = 1];

    // JWT token sent to the backend.
    string jwt_audience = 2 [(validate.rules).string.min_bytes = 1];

    // Token cluster address to fetch JWT token.
    string token_cluster = 3 [(validate.rules).string.min_bytes = 1];
  }

message FilterConfig {
    // A list of backend auth rules.
    repeated BackendAuthRule rules = 1;
}